### PR TITLE
viz: add a retry check for core control-plane pods before install

### PR DESCRIPTION
**Pull request by Pothulapati**
Fixes https://github.com/linkerd/linkerd2/issues/5426

This commit adds a new check so that `viz install` waits till
the control-plane pods are up. This follows the same pre-command
wait check model that we use for `dashboard`, etc commands.

Signed-off-by: Tarun Pothulapati <tarunpothulapati@outlook.com>

### CODE REVIEW

**Comment by adleong** [comment]
@Pothulapati it seems like there's 3 separate changes bundled together here:

* Remove the prometheus check in controller's SelfCheck method
* Some changes around waiting for the proxy-injector to start up
  * (I don't really understand those changes)
* Adding some conditions in linkerd-viz install to wait for the proxy-injector

It is possible to just address the first point in this PR to fix `linkerd check` and split the other changes out into a different PR?

**Comment by Pothulapati** [comment]
@adleong Makes sense!

**Review by adleong — APPROVED** [review]

File: viz/cmd/install.go [hunk 1]
```diff
@@ -155,3 +164,51 @@ func render(w io.Writer, valuesOverrides map[string]interface{}) error {
...
+
+		checkCmd := "linkerd check"
+		if controlPlaneNamespace != defaultLinkerdNamespace {
+			checkCmd += fmt.Sprintf(" --linkerd-namespace %s", controlPlaneNamespace)
+		}
+		fmt.Fprintf(os.Stderr, "Validate the install with: %s\n", checkCmd)
+
+		os.Exit(1)
+	}
+}
```
Comment (kleimkuhler) [inline viz/cmd/install.go:214]: Not a blocker, but seems like something maybe worth putting in `pkg/`? This is the same as what is in `public_api.go`, so worth considering.

**Review by kleimkuhler — APPROVED** [review]

File: viz/cmd/install.go (hunk 1 above)
Comment (Pothulapati) [reply viz/cmd/install.go:214]: Yep, Definitely. There are other PRs having this same fn. Will update!

**Pull request merged**